File: myblog-server/src/main/java/com/site/blog/service/impl/CategoryServiceImpl.java
package com.site.blog.service.impl;

import com.baomidou.mybatisplus.core.conditions.update.LambdaUpdateWrapper;
import com.baomidou.mybatisplus.core.toolkit.Wrappers;
import com.baomidou.mybatisplus.extension.service.impl.ServiceImpl;
import com.site.blog.constants.SysConfigConstants;
import com.site.blog.service.BlogCategoryService;
import com.site.blog.mapper.CategoryMapper;
import com.site.blog.model.entity.BlogCategory;
import com.site.blog.model.entity.Category;
import com.site.blog.service.CategoryService;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

/**
 * <p>
 * 博客分类 服务实现类
 * </p>
 *
 */
@Service
@RequiredArgsConstructor
public class CategoryServiceImpl extends ServiceImpl<CategoryMapper, Category> implements CategoryService {


  private final   CategoryMapper categoryMapper;

    private  final BlogCategoryService blogCategoryService;
    @Transactional(rollbackFor = Exception.class)
    @Override
    public boolean clearCategory(String id) {
        Category category = categoryMapper.selectById(id);
        BlogCategory blogCategory = new BlogCategory();
                blogCategory.setCategoryId( SysConfigConstants.DEFAULT_CATEGORY.getConfigField()  );

        LambdaUpdateWrapper<BlogCategory> updateWrapper = Wrappers.<BlogCategory>lambdaUpdate()
                .eq(BlogCategory::getCategoryId, category.getCategoryId());
        blogCategoryService.update(blogCategory, updateWrapper);
        return baseMapper.deleteById(category.getCategoryId())==1;
    }
}

File: myblog-kt/src/main/kotlin/com/site/blog/service/impl/CategoryServiceImpl.kt
package com.site.blog.service.impl

import com.baomidou.mybatisplus.core.toolkit.Wrappers
import com.baomidou.mybatisplus.extension.service.impl.ServiceImpl
import com.site.blog.constants.SysConfigConstants
import com.site.blog.mapper.CategoryMapper
import com.site.blog.model.entity.BlogCategory
import com.site.blog.model.entity.Category
import com.site.blog.service.BlogCategoryService
import com.site.blog.service.CategoryService
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

/**
 *
 *
 * 博客分类 服务实现类
 *
 *
 */
@Service
class CategoryServiceImpl(
    private val categoryMapper: CategoryMapper,
    private val blogCategoryService: BlogCategoryService
) : ServiceImpl<CategoryMapper?, Category?>(), CategoryService {
    @Transactional(rollbackFor = [Exception::class])
    override fun clearCategory(id: String?): Boolean {
        val category = categoryMapper.selectById(id)
        val blogCategory = BlogCategory()
        blogCategory.categoryId = SysConfigConstants.DEFAULT_CATEGORY.configField
        val updateWrapper = Wrappers.lambdaUpdate<BlogCategory>()
            .eq(BlogCategory::categoryId, category!!.categoryId)
        blogCategoryService.update(blogCategory, updateWrapper)
        return baseMapper!!.deleteById(category.categoryId) == 1
    }
}